vm/string_test.go

```diff
@@ -76,6 +131,61 @@ func TestStringOperation(t *testing.T) {
 		{`"Three " * 3`, "Three Three Three "},
 		{`"Zero" * 0`, ""},
 		{`"Minus" * 1`, "Minus"},
+		{`"Hello"[1]`, "e"},
+		{`"Hello"[5]`, nil},
+		{`"Hello"[-1]`, "o"},
+		{`"Hello"[-6]`, nil},
+		{`"Hello\nWorld"[5]`, "\n"},
+		{`"Ruby"[1] = "oo"`, "Rooby"},
+		{`"Go"[2] = "by"`, "Goby"},
+		{`"Ruby"[-3] = "oo"`, "Rooby"},
+		{`"Hello"[-5] = "Tr"`, "Trello"},
+		{`"Hello\nWorld"[5] = " "`, "Hello World"},
+	}
+
+	for i, tt := range tests {
+		vm := initTestVM()
+		evaluated := vm.testEval(t, tt.input)
+		checkExpected(t, i, evaluated, tt.expected)
+	}
+}
+
+func TestStringOperationFail(t *testing.T) {
+	vm := initTestVM()
+	testsFail := []struct {
+		input    string
+		expected *Error
+	}{
+		{`"Taipei" + 101`, initErrorObject(TypeErrorClass, "Expect argument to be String. got=%T", vm.initIntegerObject(101))},
```

Comment:
Here you should write down the whole error messages. Because if there's an issue in `initErrorObject` you'll never find out in these tests. So the `expected` field's type should be string`.